### PR TITLE
Update to typed `jsontokens` lib

### PR DESCRIPTION
**Pull request by zone117x**
Update the to the v2 version of `jsontokens` which is ported to Typescript, and its dependency `key-encoder`:
https://github.com/blockstack/key-encoder-js
https://github.com/blockstack/jsontokens-js

### CODE REVIEW

File: src/auth/authSession.ts [hunk 1]
```diff
@@ -135,33 +135,34 @@ export function getCoreSession(coreHost: string,
     return Promise.reject('No authRequest provided')
   }
 
-  let payload = null
-  let authRequestObject = null
   try {
-    authRequestObject = decodeToken(authRequest)
+    const authRequestObject = decodeToken(authRequest)
```
Comment (yknl) [inline src/auth/authSession.ts:139]: Core sessions are no longer used, I think we should remove this code in a separate PR.

**Review by yknl — APPROVED** [review]

**Review by timstackblock — APPROVED** [review]

**Comment by timstackblock** [comment]
What functionality does this directly impact, or is a regression test of general functionality only needed?

**Comment by yknl** [comment]
This PR doesn't change any functionality. A regression test of the auth flow with this branch of blockstack.js in both the app and browser is what's needed.

**Comment by timstackblock** [comment]
Thanks @yknl starting that now

**Comment by timstackblock** [comment]
When trying to sign in with blockstack browser in the app I run into this issue please take a look at this, I am moving this issue back to in progress.  

Video of the issue
https://screencast-o-matic.com/watch/cqjlI8ORFv

Screen Shot 2019-08-12 at 9 57 28 AM

**Comment by zone117x** [comment]
@timstackblock Server connectivity error looks unrelated. Can you verify the process you are using for testing the Browser with this blockstack.js branch?

**Comment by timstackblock** [comment]
This works the error I was getting was from running in development mode and it happens on this branch and master its unrelated.

**Comment by zone117x** [comment]
Okay great, were you able to verify that the browser e2e tests pass when the browser is using this branch?

**Comment by timstackblock** [comment]
Yes just ran the test locally and we are good to go

**Pull request merged**